tinygrad/lazy.py

```diff
@@ -105,7 +105,6 @@ def e(self, op:Union[LoadOps, UnaryOps, BinaryOps, TernaryOps], *in_srcs:LazyBuf
         srcs.append(s)
     assert all_same(dts:=[x.dtype.scalar() for x in (srcs[1:] if op is TernaryOps.WHERE else srcs)]), f"all dtypes must match {dts} on {op}"
     assert all_same([x.shape for x in srcs]), f"all shapes must be the same {[x.shape for x in srcs]}"
-    if op is TernaryOps.WHERE: assert srcs[0].dtype == dtypes.bool, "TernaryOps.WHERE must have the first arg be bool"
```

Comment:
I think this can be removed cuz 
`return mlops.Where.apply(x.cast(dtypes.bool), *y._broadcasted(z))` already pins `x` to `bool`

I went down git blame and I think this was added to cover for WEBGPU hack
and WEBGPU is rip

Reply:
let's keep this. it completes the assert check two lines above